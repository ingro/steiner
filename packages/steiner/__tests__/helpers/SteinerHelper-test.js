import SteinerHelper from '../../src/index';

describe('SteinerHelper', () => {
    it('is instantiable', () => {
        const helper = new SteinerHelper();

        expect(helper).toBeInstanceOf(SteinerHelper);
    });

    describe('getCreateActionsOptions', () => {
        it('returns undefined if no helper options are defined', () => {
            const helper = new SteinerHelper();

            expect(helper.getCreateActionsOptions()).toEqual({});
        }); 

        it('returns actionMessageTemplates options if defined as messageTemplates', () => {
            const helper = new SteinerHelper({
                actionMessageTemplates: {
                    createFail: '${resource} cannot be created!',
                    createSuccess: '${resource} created!'
                }
            });

            expect(helper.getCreateActionsOptions()).toEqual({
                messageTemplates: {
                    createFail: '${resource} cannot be created!',
                    createSuccess: '${resource} created!'
                }
            });
        });

        it('returns actionMessages options if defined as messages', () => {
            const helper = new SteinerHelper({
                actionMessages: {
                    createFail: 'Hooray!',
                    createSuccess: 'Boooo'
                }
            });

            expect(helper.getCreateActionsOptions()).toEqual({
                messages: {
                    createFail: 'Hooray!',
                    createSuccess: 'Boooo'
                }
            });
        });

        it('allow to override the helper\'s options', () => {
            const helper = new SteinerHelper({
                actionMessageTemplates: {
                    createFail: '${resource} cannot be created!',
                    createSuccess: '${resource} created!'
                }
            });

            const options = {
                messageTemplates: {
                    deleteFail: '${resource} cleared!'
                }
            };

            expect(helper.getCreateActionsOptions(options)).toEqual({
                messageTemplates: {
                    deleteFail: '${resource} cleared!'
                }
            });
        });
    });
});