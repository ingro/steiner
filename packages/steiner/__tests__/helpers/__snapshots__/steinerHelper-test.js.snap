exports[`steinerHelper createDefaultState returns the custom state 1`] = `
Object {
  "current": Object {
    "errorMessage": null,
    "isFetching": false,
    "item": null,
  },
  "list": Object {
    "errorMessage": null,
    "filters": Object {
      "orderDirection": "ASC",
      "orderKey": null,
      "page": 1,
      "perPage": 50,
      "q": "",
    },
    "isFetching": false,
    "itemsById": Object {},
    "itemsId": Array [],
    "selected": Array [],
    "total": 0,
  },
}
`;

exports[`steinerHelper createDefaultState returns the default state defined in the reducer 1`] = `
Object {
  "current": Object {
    "errorMessage": null,
    "isFetching": false,
    "item": null,
  },
  "list": Object {
    "errorMessage": null,
    "filters": Object {
      "orderDirection": "ASC",
      "orderKey": null,
      "page": 1,
      "perPage": 20,
      "q": "",
    },
    "isFetching": false,
    "itemsById": Object {},
    "itemsId": Array [],
    "selected": Array [],
    "total": 0,
  },
}
`;

exports[`steinerHelper getCreateActionsOptions allow to override the helper's options 1`] = `
Object {
  "messageTemplates": Object {
    "deleteFail": "{{resource}} cleared!",
  },
  "messages": undefined,
  "notificationTitles": Object {
    "fail": "Oh snap!",
    "success": "Hooray!",
  },
}
`;

exports[`steinerHelper getCreateActionsOptions returns actionMessageTemplates options if defined as messageTemplates 1`] = `
Object {
  "messageTemplates": Object {
    "createFail": "{{resource}} cannot be created!",
    "createSuccess": "{{resource}} created!",
  },
  "messages": undefined,
  "notificationTitles": Object {
    "fail": "Oh snap!",
    "success": "Hooray!",
  },
}
`;

exports[`steinerHelper getCreateActionsOptions returns actionMessages options if defined as messages 1`] = `
Object {
  "messageTemplates": Object {
    "createFail": "An error occured while creating {{resource}}",
    "createSuccess": "{{resource}} created with success!",
    "deleteFail": "An error occured while deleting {{resource}}",
    "deleteSuccess": "{{resource}} deleted with success!",
    "updateFail": "An error occured while updating {{resource}}",
    "updateSuccess": "{{resource}} updated successfully!",
  },
  "messages": Object {
    "createFail": "Hooray!",
    "createSuccess": "Boooo",
  },
  "notificationTitles": Object {
    "fail": "Oh snap!",
    "success": "Hooray!",
  },
}
`;

exports[`steinerHelper getCreateActionsOptions returns translated custom messages 1`] = `
Object {
  "messageTemplates": Object {
    "createFail": "Booo {{resource}}",
    "createSuccess": "Hooray {{resource}}",
  },
  "messages": undefined,
  "notificationTitles": Object {
    "fail": "Fail",
    "success": "Success",
  },
}
`;

exports[`steinerHelper getCreateActionsOptions returns translated default messages 1`] = `
Object {
  "messageTemplates": Object {
    "createFail": "Si è verificato un errore inaspettato durante la creazione di {{resource}}",
    "createSuccess": "{{resource}} creato con successo!",
    "deleteFail": "Si è verificato un errore inaspettato durante la cancellazione di {{resource}})",
    "deleteSuccess": "{{resource}} cancellato con successo!",
    "updateFail": "Si è verificato un errore inaspettato durante l\'aggiornamento di {{resource}}",
    "updateSuccess": "{{resource}} aggiornato con successo!",
  },
  "messages": undefined,
  "notificationTitles": Object {
    "fail": "Error!",
    "success": "Successo!",
  },
}
`;

exports[`steinerHelper getCreateActionsOptions returns undefined if no helper options are defined 1`] = `
Object {
  "messageTemplates": Object {
    "createFail": "An error occured while creating {{resource}}",
    "createSuccess": "{{resource}} created with success!",
    "deleteFail": "An error occured while deleting {{resource}}",
    "deleteSuccess": "{{resource}} deleted with success!",
    "updateFail": "An error occured while updating {{resource}}",
    "updateSuccess": "{{resource}} updated successfully!",
  },
  "messages": undefined,
  "notificationTitles": Object {
    "fail": "Oh snap!",
    "success": "Hooray!",
  },
}
`;
